gpm_api/tests/io/test_checks.py

```diff
@@ -0,0 +1,447 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Jul 17 15:41:14 2023
+
+@author: ghiggi
+"""
+
+import pytest
+import datetime
+import numpy as np
+import platform
+from gpm_api.io import checks
+from gpm_api.io.products import available_products, available_scan_modes
+
+
+def test_is_not_empty() -> None:
+    """Test is_not_empty() which always returns a boolean"""
+
+    # Test False responses:
+    for obj in [None, (), {}, []]:
+        res = checks.is_not_empty(obj)
+        assert res is False, "Function returned True, expected False"
+
+    # Test True responses:
+    for obj in [[1, 2, 3], (1, 2, 3), (1), [1]]:
+        res = checks.is_not_empty(obj)
+        assert res is True, "Function returned False, expected True"
+
+
+def test_is_empty() -> None:
+    """Test is_empty()"""
+
+    # Test False responses:
+    for obj in [[1, 2, 3], (1, 2, 3), (1), [1]]:
+        res = checks.is_empty(obj)
+        assert res is False, "Function returned True, expected False"
+
+    # Test True responses:
+    for obj in [None, (), {}, []]:
+        res = checks.is_empty(obj)
+        assert res is True, "Function returned False, expected True"
+
+
+def test_check_base_dir() -> None:
+    """Check path constructor for base_dir"""
+
+    # Check leading slash is removed
+    res = checks.check_base_dir("/home/user/gpm/")
+    assert res == "/home/user/gpm", "Leading slash is not removed"
+
+    # Check leading slash is removed
+    res = checks.check_base_dir("/home/user/gpm")
+    assert res == "/home/user/gpm", "Leading slash is not removed"
+
+    # Check if GPM, it is removed
+    res = checks.check_base_dir("/home/user/gpm/GPM")
+    assert res == "/home/user/gpm", "GPM is not removed"
+
+    # # Check windows path
+    # res = checks.check_base_dir("C:\\home\\user\\gpm\\GPM")
+    # assert res == "C:\\home\\user\\gpm", "GPM is not removed"
```

Comment:
Not needed if using [this](https://github.com/ghiggi/gpm_api/pull/8#discussion_r1270572752)